packages/next/src/client/components/app-router.tsx

```diff
@@ -362,14 +362,25 @@ function Router({
       forward: () => window.history.forward(),
       prefetch: (href, options) => {
         // Don't prefetch for bots as they don't navigate.
+        if (isBot(window.navigator.userAgent)) {
+          return
+        }
+
+        let url: URL
+        try {
+          url = new URL(addBasePath(href), window.location.href)
+        } catch (err) {
+          console.error(
```

Comment:
We should save the error, in dev mode if it's invalid url we should throw the error so users can discover earlier. And in production build we just swallow the error, not doing any prefetch and also error in console